.github/workflows/msbuild.yml

```
@@ -51,7 +54,7 @@ jobs:
 
   integration-tests:
     needs: check-deep-tests
-    if: always() && (needs.check-deep-tests.result == 'success' || contains(github.event.pull_request.labels.*.name, 'run-deep-tests'))
+    if: always() && (needs.check-deep-tests.result == 'success' || contains(github.event.pull_request.labels.*.name, 'run-deep-tests') || (github.repository != 'dafny-lang/dafny'  && vars.TEST_ON_FORK == 'true' ))
```

```suggestion
    if: always() && (needs.check-deep-tests.result == 'success' || contains(github.event.pull_request.labels.*.name, 'run-deep-tests') || vars.TEST_ON_FORK == 'true' )
```
I think it's fine to just not set the variable.

---

Not setting the variable means it is an empty string. But I think you do have to set it to something to enable the behavior.

As for non-PR pushes on the main repo. I did not think that was a recommended workflow. But if it is being used, by your proposal, the failure of nightly testing would turn off testing, as long as one did not run on the new variable
We could add a guard on the rpo name as well.